akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -1452,6 +1452,20 @@ trait FlowOps[+Out, +Mat] {
   def idleTimeout(timeout: FiniteDuration): Repr[Out] = via(new Timers.Idle[Out](timeout))
 
   /**
+    * If the time between emission of element and following backpressure notification exceed the provided timeout,
```

Comment:
Makes a lot of sense - `keepAlive` is a little bit different though (opposite). I was not correct stating `flow.keepAlive.map` can replace this stage. I've verified it today by running tests against it and border cases failed. The difference is described in #20131 - we timeout on lack of backpressure not on lack of elements. Following cases fails for me if `flow.keepAlive.map` used:

- not throw if publisher is less frequent then timeout
- not throw if publisher won't perform emission ever
- throw if subscriber won't notify backpressure on time
- not throw if publisher completes without fulfilling subscriber's demand

Cause is the same for all of them. `keepAlive` pushes `inject()` and `map` subscribed to value produced by `inject()` throws/fails. I think that is the difference - stage under review is meant not to fail if backpressure was notified but producer is too slow or will never deliver next element.

The real question is if stage which fails on "Hey, why are you not backpressuring me! I quit!" can be combined differently or has a value at all :-).